### PR TITLE
Added modal component

### PR DESCRIPTION
**Pull request by kiranpratyush**
The modal component is displayed on the screen while adding videos to the playlist.

1. styling of add to playlist modal is not perfect yet
2. code is not fully yet refactored
3. Logo is of youtube but will be changed
4. Check box is not properly styled

https://user-images.githubusercontent.com/77445957/161370729-8a6f1f3e-eb33-42fd-b796-1844817184bf.mp4

[Live Link](https://spiffy-fenglisu-8e9c7c.netlify.app/)

### CODE REVIEW

File: src/Contexts/PlayLists.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      if (!previousStateCopy[playListName]) {
+        previousStateCopy[playListName] = [id];
+      } else {
+        previousStateCopy[playListName].push(id);
+      }
+      setPlayListData(playListName, previousStateCopy[playListName]);
+      return previousStateCopy;
+    case 'REMOVE_FROM_PLAYLIST':
+      const copyOfPreviousState = { ...previousState };
+      const indexOfTheId = copyOfPreviousState[payload.name].findIndex(
```
Comment (gopalbbharadva) [inline src/Contexts/PlayLists.js:21]: Suggestion : You can destructure payload's data outside switch , this looks cleaner your each cases.

**Review by gopalbbharadva — APPROVED** [review]
Modal looks Nice,
Improvement: if not then , when user clicks outside of modal and menu , both should close.

File: src/Components/AddToPlayists/AddPlayList.css [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  z-index: 200;
+  top: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.addtoplaylist__maincontainer{
+  background-color: black;
+   padding:20px 20px;
```
Comment (bipin7yadav) [inline src/Components/AddToPlayists/AddPlayList.css:15]: you can use rem instead of px

File: src/Components/AddToPlayists/AddPlayList.css [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+   color:white;
+}
+.addtoplaylist__container {
+  
+ 
+  color: white;
```
Comment (bipin7yadav) [inline src/Components/AddToPlayists/AddPlayList.css:21]: you can use color variables

**Review by bipin7yadav — APPROVED** [review]

File: src/Contexts/PlayLists.js (hunk 1 above)
Comment (kiranpratyush) [reply src/Contexts/PlayLists.js:21]: Yes Thank you

**Pull request merged**